x-pack/plugins/observability_solution/profiling/public/plugin.tsx

```diff
@@ -64,29 +65,47 @@ export class ProfilingPlugin
     ];
 
     const kuerySubject = new BehaviorSubject<string>('');
+    const appUpdater$ = new BehaviorSubject<AppUpdater>(() => ({}));
 
     const section$ = combineLatest([from(coreSetup.getStartServices()), kuerySubject]).pipe(
       map(([[coreStart], kuery]) => {
         if (coreStart.application.capabilities.profiling.show) {
-          const sections: NavigationSection[] = [
-            {
-              label: i18n.translate('xpack.profiling.navigation.sectionLabel', {
-                defaultMessage: 'Universal Profiling',
-              }),
-              entries: links.map((link) => {
-                return {
-                  app: OBLT_PROFILING_APP_ID,
-                  label: link.title,
-                  path: `${link.path}?kuery=${kuery ?? ''}`,
-                  matchPath: (path) => {
-                    return path.startsWith(link.path);
-                  },
-                };
-              }),
-              sortKey: 700,
-            },
-          ];
-          return sections;
+          let isSidebarEnabled = true;
+          coreStart.chrome
+            .getChromeStyle$()
+            .pipe(take(1))
+            .subscribe((style) => (isSidebarEnabled = style === 'classic'));
+
+          if (isSidebarEnabled) {
+            // classic navigation
+            const sections: NavigationSection[] = [
+              {
+                label: i18n.translate('xpack.profiling.navigation.sectionLabel', {
+                  defaultMessage: 'Universal Profiling',
+                }),
+                entries: links.map((link) => {
+                  return {
+                    app: OBLT_PROFILING_APP_ID,
+                    label: link.title,
+                    path: `${link.path}?kuery=${kuery ?? ''}`,
+                    matchPath: (path) => {
+                      return path.startsWith(link.path);
+                    },
+                  };
+                }),
+                sortKey: 700,
+              },
+            ];
+            return sections;
+          } else {
+            // solution navigation
+            appUpdater$.next(() => ({
+              deepLinks: links.map((link) => ({
+                ...link,
+                path: `${link.path}?kuery=${encodeURIComponent(kuery) ?? ''}`,
```

Comment:
Same here, also do we want to persist any other params coming from `link` or that's not the case here? I mean having the `?kuery=....&other params from link if any`

Reply:
I doubt there is need for it. The complaint was mostly about missing `kuery` param using solution navigation. The date range parameters are in place (they must be handled separately).